packages/@sanity/form-builder/src/inputs/InvalidValueInput/InvalidValueInput.tsx

```diff
@@ -73,29 +71,107 @@ export default class InvalidValueInput extends React.PureComponent<InvalidValueP
+        <Button onClick={handleClearClick} tone="critical" text="Reset value" />
+      </Stack>
     )
 
-    return <Warning heading="Content has invalid type" message={message} />
+    return (
+      <Alert status="error" suffix={suffix} title={<>Invalid property value</>}>
+        <Text as="p" muted size={1}>
+          The property value is stored as a value type that does not match the expected type.
+        </Text>
+
+        <Details marginTop={4} open={__DEV__} title={<>Developer info</>}>
+          <Stack space={3}>
+            {validTypes.length === 1 && (
+              <Text as="p" muted size={1}>
+                The value of this property must be of type <code>{validTypes[0]}</code> according to
+                the schema.
+              </Text>
+            )}
+
+            {validTypes.length === 1 && (
+              <Text as="p" muted size={1}>
+                Mismatching value types typically occur when the schema has recently been changed.
+              </Text>
+            )}
+
+            {validTypes.length !== 1 && (
+              <Text as="p" muted size={1}>
+                Only the following types are valid here according to schema:
+              </Text>
+            )}
+
+            {validTypes.length !== 1 && (
+              <ul>
+                {validTypes.map((validType) => (
+                  <Text as="li" key={validType}>
+                    <code>{validType}</code>
+                  </Text>
+                ))}
+              </ul>
+            )}
+
+            <Stack marginTop={2} space={2}>
+              <Text size={1} weight="semibold">
+                The current value (<code>{actualType}</code>)
+              </Text>
+
+              <Card border padding={2} radius={2} tone="inherit">
+                <Code language="json" size={1}>
+                  {JSON.stringify(value, null, 2)}
+                </Code>
+              </Card>
+            </Stack>
+
+            {converters.length > 0 && (
+              <Stack space={1}>
+                {converters.map((converter, converterIndex) => (
+                  <ConvertButton
+                    converter={converter}
+                    // eslint-disable-next-line react/no-array-index-key
+                    key={converterIndex}
+                    onConvert={handleConvertTo}
+                    value={value}
+                  />
+                ))}
+              </Stack>
+            )}
+          </Stack>
+        </Details>
+      </Alert>
+    )
   }
+)
+
+InvalidValueInput.displayName = 'InvalidValueInput'
+
+function ConvertButton({
+  converter,
+  onConvert,
+  value,
+}: {
+  converter: Converter
+  onConvert: (v: string | number | boolean | Record<string, unknown>) => void
+  value: unknown
+}) {
+  const handleClick = useCallback(() => onConvert(converter.convert(value)), [
+    converter,
+    onConvert,
+    value,
+  ])
+
+  return (
+    <Button
+      key={`${converter.from}-${converter.to}`}
```

Comment:
Is this `key` property necessary now?

Reply:
Don't think so! It was replaced with an index-key on the ConvertButton usage here: https://github.com/sanity-io/sanity/commit/8a556b9d18fe49ea65cc638f5c5b31df881f7335#diff-6bd9775a6ced1a969930ca6a5f86b7bee8c62e969d53fa720c836a58418fcfa5R135

I'm not sure why that was changed to an index key instead of keeping it as-is(?)

Reply:
Fixed in https://github.com/sanity-io/sanity/commit/2f3ac838e39f57cf74f3a3d68bbd2ead842c7815